sql/core/src/main/scala/org/apache/spark/sql/execution/command/ddl.scala

```diff
@@ -636,7 +632,6 @@ case class AlterTableRecoverPartitionsCommand(
     val catalog = spark.sessionState.catalog
     val table = catalog.getTableRawMetadata(tableName)
     val tableIdentWithDB = table.identifier.quotedString
-    DDLUtils.verifyAlterTableType(catalog, table, isView = false)
```

Comment:
This is safe to remove after #31403.

Reply:
no, it doesn't. need to merge https://github.com/apache/spark/pull/31403 firstly.